### PR TITLE
[t2178] timeline follows currenTime

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7914a78eafd6c2d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/7914a78eafd6c2d/output.txt

Total script time: 0.99 mins

Server is running on http://pr912.butyr.org/templates/basic/
Run the Test Suite: http://pr912.butyr.org/test/

File: src/timeline/super-scrollbar.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
       e.preventDefault();
       e.stopPropagation();
       _offset = e.clientX - _rect.left - _viewPort.offsetLeft;
+      _media.pause();  // pause the media here to diffuse confusiong with scrolling & playing
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js]: typo on confusion.

File: src/timeline/super-scrollbar.js [hunk 2]
```diff
@@ -92,6 +93,7 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
     onLeftMouseDown = function( e ) {
       e.preventDefault();
       e.stopPropagation();
+      _media.pause();  // pause the media here to diffuse confusiong with scrolling & playing
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js]: Same typo.

File: src/timeline/super-scrollbar.js [hunk 3]
```diff
@@ -100,6 +102,7 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
     onRightMouseDown = function( e ) {
       e.preventDefault();
       e.stopPropagation();
+      _media.pause();  // pause the media here to diffuse confusiong with scrolling & playing
```
Comment (ScottDowne) [inline src/timeline/super-scrollbar.js]: Same typo.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/46e689016b3faf1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/02dd6b9df0ac7ba/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/46e689016b3faf1/output.txt

Total script time: 1.06 mins

Server is running on http://pr912.butyr.org/templates/basic/
Run the Test Suite: http://pr912.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/02dd6b9df0ac7ba/output.txt

Total script time: 2.27 mins

File: src/timeline/media.js [hunk 4]
```diff
@@ -51,12 +51,27 @@ define( [ "core/trackevent", "core/track", "core/eventmanager",
       _tracksContainer.update();
     }
 
-    function snapToCurrentTime(){
-      _tracksContainer.snapTo( _media.currentTime );
+    function onMediaTimeUpdate() {
+      // Move the viewport to be centered around the scrubber
+      _tracksContainer.followCurrentTime();
+      // Align the timebar again to remove jitter
+      // TODO: this is expensive, and only fixes 50% of the problem
```
Comment (ScottDowne) [inline src/timeline/media.js:58]: Make sure this is filed.

Comment (secretrobotron) [inline src/timeline/media.js:58]: It is. https://webmademovies.lighthouseapp.com/projects/65733/tickets/2213

**Pull request merged**